### DISCUSSION TITLE
Can't seem to figure out aligning of columns

### DISCUSSION THREAD

**Discussion by SwayKh** [saghen/blink.cmp#414] [General]
So, I like to have a bigger min_width of my cmp menu, so that the menu doesn't keep bouncing around in sizes based on the length of suggestion.

According to the README, you can align the columns with `width.fill` property of the component, but it doesn't seem to be working for me.

This is my config
```lua
      min_width = 35,
      draw = {
        columns = { { "kind_icon" }, { "label", "label_description", gap = 1 }, { "kind" } },
        components = {
          kind = { width = { fill = false } },
          kind_icons = { width = { fill = false } },
          label = { width = { fill = true } },
          label_description = { width = { fill = true } },
        },
      },
```

With this config, what I think it should be like is, having "kind_icons" "label" "label_description" on one side of the menu, and have the "kind" component should align with the right side. Or Am I doing something wrong.

But right now, the menu is just like regular menu, with everything aligned on left side, and a huge chunk of empty space on the right. But I want the chunk of space to be between label and kind components.

**Comment by saghen** [comment]
Setting `width.fill` effectively makes the components that come afterwards, in the same column, right aligned. So you likely want

```lua
draw = {
  columns = { { "kind_icon" }, { "label", "label_description", "kind", gap = 1 } },
  components = {
    label = { width = { fill = false } }, -- default is true
    label_description = { width = { fill = true } },
  },
},
```

Although, I'm not sure we handle the `min_width` property for filling. If it's still not working, open up an issue and I can take a look

**Reply by SwayKh** [reply]
No, That didn't seem to fix it. It still look like this, with the space on the side.

![1733159105_grim](https://github.com/user-attachments/assets/1586f328-2c7a-4016-8b2f-10daecc972e3)

I've created an issue, like you suggested

**Reply by saghen** [reply]
For people looking for a solution, please follow/check #424
